lib/srv/db/common/errors.go

```diff
@@ -47,29 +47,39 @@ func ConvertError(err error) error {
 		return nil
 	}
 	// Unwrap original error first.
-	if _, ok := err.(*trace.TraceErr); ok {
+	var traceErr *trace.TraceErr
+	if errors.As(err, &traceErr) {
 		return ConvertError(trace.Unwrap(err))
 	}
-	if pgErr, ok := err.(pgError); ok {
+	var pgErr pgError
+	if errors.As(err, &pgErr) {
 		return ConvertError(pgErr.Unwrap())
 	}
-	if causer, ok := err.(causer); ok {
-		return ConvertError(causer.Cause())
+
+	var c causer
+	if errors.As(err, &c) {
+		return ConvertError(c.Cause())
 	}
 	if _, ok := status.FromError(err); ok {
 		return trail.FromGRPC(err)
 	}
-	switch e := trace.Unwrap(err).(type) {
-	case *googleapi.Error:
-		return convertGCPError(e)
-	case awserr.RequestFailure:
-		return awslib.ConvertRequestFailureError(e)
-	case *azcore.ResponseError:
-		return azurelib.ConvertResponseError(e)
-	case *pgconn.PgError:
-		return convertPostgresError(e)
-	case *mysql.MyError:
-		return convertMySQLError(e)
+
+	var googleAPIErr *googleapi.Error
+	var awsRequestFailureErr awserr.RequestFailure
+	var azResponseErr *azcore.ResponseError
+	var pgError *pgconn.PgError
+	var myError *mysql.MyError
```

Comment:
This is a bit sad, as each errors.As invocation is a separate traversal of the error chain. Hopefully the chains are small enough that this won't make much of a difference.